x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/TrainedModelAssignment.java

```diff
@@ -153,6 +156,43 @@ public String[] getStartedNodes() {
             .toArray(String[]::new);
     }
 
+    public Optional<String> selectRandomStartedNodeWeighedOnAllocations() {
+        List<String> nodeIds = new ArrayList<>(nodeRoutingTable.size());
+        List<Integer> cumulativeAllocations = new ArrayList<>(nodeRoutingTable.size());
+        int allocationSum = 0;
+        for (Map.Entry<String, RoutingInfo> routingEntry : nodeRoutingTable.entrySet()) {
+            if (RoutingState.STARTED.equals(routingEntry.getValue().getState())) {
+                nodeIds.add(routingEntry.getKey());
+                allocationSum += routingEntry.getValue().getCurrentAllocations();
+                cumulativeAllocations.add(allocationSum);
+            }
+        }
```

Comment:
OK, I am wondering here, since we do the additional check regarding to previous versions, why can't we weigh old nodes with started deployments with an allocation value of 1?